central/serviceaccount/datastore/datastore_impl.go

```diff
@@ -35,11 +36,25 @@ func (d *datastoreImpl) GetServiceAccount(ctx context.Context, id string) (*stor
 }
 
 func (d *datastoreImpl) SearchRawServiceAccounts(ctx context.Context, q *v1.Query) ([]*storage.ServiceAccount, error) {
-	return d.searcher.SearchRawServiceAccounts(ctx, q)
+	serviceAccounts := make([]*storage.ServiceAccount, 0, paginated.GetLimit(q.GetPagination().GetLimit(), whenUnlimited))
+	err := d.storage.GetByQueryFn(ctx, q, func(serviceAccount *storage.ServiceAccount) error {
+		serviceAccounts = append(serviceAccounts, serviceAccount)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return serviceAccounts, nil
 }
 
 func (d *datastoreImpl) SearchServiceAccounts(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
-	return d.searcher.SearchServiceAccounts(ctx, q)
+	serviceAccounts, results, err := d.searchServiceAccounts(ctx, q)
```

Comment:
`searchServiceAccounts` still has to query the data twice. A good followup to eliminating searchers could be to eliminate use of these `v1.SearchResult` functions. Mostly `search.Result` should have all the relevant information.

Reply:
yeah.  We could just get the objects themselves and then fill in the `SearchResult`.  Felt out of scope because we have to get the highlighting.  We'll keep iterating on these topics and try to get this stuff out and simplified.


https://issues.redhat.com/browse/ROX-26633